### ISSUE TITLE
[Feature] Add sorting and filtering to the partners page

### ISSUE THREAD

**Issue by dtaivpp** [opensearch-project/project-website#1693]
**Is your feature request related to a problem? Please describe.**
At the moment it can be challenging to find information on partners you may be interested in. We have a wide range of partners from consultants, managed service providers (MSP's), and those extending OpenSearch. 

**Describe the solution you'd like**
There should be a mechanism to filter the partner list to just the types you are looking for using the existing metadata. This may be a drop down menu or picklist. 

**Describe alternatives you've considered**
We could also just add tags to partners cards. This solution would solve the problem temporally but would not scale well with our partner list.

**Comment by krisfreedain** [comment]
Yes - definitely planned to take this on as part of the new partner page experience - adding to the Website Update Project / backlog while we work on data normalization in https://github.com/opensearch-project/project-website/issues/1345
